pkg/reconciler/event.go

```diff
@@ -24,6 +24,11 @@ func (r *Reconciler) detectProvider(ctx context.Context, pr *v1beta1.PipelineRun
 
 	event := buildEventFromPipelineRun(pr)
 
+	// for testing, if already initialised then use it
+	if r.provider != nil {
```

Comment:
usually having a condition for testing in code is pretty bad 🙃

Reply:
can't you use the interface to fake the testing?

Reply:
I was trying but couldn't figure out how to in the current implementation

Reply:
refactored the code little bit
moved detect provide outside. it is tested separately